src/features/providers/linter/lint.ts

```diff
@@ -125,7 +125,7 @@ export default class LintingProvider {
     const workspaceFolder = vscode.workspace.workspaceFolders
       ? vscode.workspace.workspaceFolders[0].uri.fsPath
       : undefined;
-    const rootPath = workspaceFolder ? Utilities.normalizePath(workspaceFolder) : undefined;
+    const rootPath = workspaceFolder ? Utilities.normalizePath(workspaceFolder) : Utilities.normalizePath(".");
     const workingDirectory = workspacePath ?? Configuration.workingDirectory(rootPath);
     const filePath = document && !document.isUntitled ? Utilities.normalizePath(document.fileName) : workingDirectory;
```

Comment:
I think the issue might actually be the filePath should be using the document.fileName, but for some reason the `document.isUntitled` is `true` sometimes in workspaceless instances even though the file has a filename. I am going to update this to instead check `document.fileName`.

This change will be released in v3.1.2, let me know if there are still issues and I can look into it further.

Reply:
I think this is an ok solution, my only concern is the `--stdin-filename` path becomes something like `sqlfluff lint --format json --stdin-filename ../../../../../path/to/vscode/Untitled-1 -`. By itself, this is fine as it will pick up the workspace's root `.sqlfluff` file but could have unintended side effects if any additional `.sqlfluff` files exist on that path.